strawberry/schema/types/base_scalars.py

```diff
@@ -2,12 +2,25 @@
 import decimal
 import uuid
 from operator import methodcaller
-
+from typing import Callable
 import dateutil.parser
+from graphql import GraphQLError
 
 from strawberry.custom_scalar import scalar
 
 
+def wrap_iso_parser(parser: Callable) -> Callable:
+    def inner(value: str):
+        try:
+            return parser(value)
+        except ValueError as e:
+            raise GraphQLError(
+                f'Expected ISO formatted string, received "{value}". {e}'
```

Comment:
I wanted to make it similar to graphql-core. Maybe it can be a bit more similar:
`"Expected value of type DateTime', found {value}."`
from https://github.com/graphql-python/graphql-core/blob/03c2b06e4636ed8a89c7a4c7e56d244fd9d00bde/src/graphql/validation/rules/values_of_correct_type.py#L93

Reply:
That could be a bit misleading though because you're not expecting a "value of type DateTime", you're expecting something that can represent a DateTime type.

Reply:
I think you actually do expect `DateTime` here, as in the schema it will be `field: DateTime`. Scalar errors are raised within schema context/logic, so I'd say expecting type `DateTime` makes a lot of sense.

Reply:
What about `Value cannot represent a DateTime: "{value}". {e}`?